transport_plugins/bled112/iotile_transport_bled112/bled112_cmd_co.py

```diff
@@ -0,0 +1,843 @@
+            return False, {'reason': 'Timeout waiting for command response'}
+
+        handle, result = unpack("<BH", response.payload)
+        if result != 0:
+            return False, None
+
+        events = self._wait_process_events(0.5, event_filter_func, end_filter_func)
+        gatt_events = [x for x in events if event_filter_func(x)]
+        end_events = [x for x in events if end_filter_func(x)]
+
+        if len(end_events) == 0:
+            return False, None
+
+        #Make sure we successfully probed the gatt table
+        end_event = end_events[0]
+        _, result, _ = unpack("<BHH", end_event.payload)
+        if result != 0:
+            self._logger.warn("Error enumerating GATT table, protocol error code = %d (0x%X)" % (result, result))
+            return False, None
+
+        services = {}
+        for event in gatt_events:
+            process_gatt_service(services, event)
+
+        return True, {'services': services}
+
+    def _probe_characteristics(self, conn, services, timeout=5.0):
+        """Probe gatt services for all associated characteristics in a BLE device
+
+        Args:
+            conn (int): the connection handle to probe
+            services (dict): a dictionary of services produced by probe_services()
+            timeout (float): the maximum number of seconds to spend in any single task
+        """
+
+        for service in services.values():
+            success, result = self._enumerate_handles(conn, service['start_handle'],
+                                                      service['end_handle'])
+
+            if not success:
+                return False, None
+
+            attributes = result['attributes']
+
+            service['characteristics'] = {}
+
+            last_char = None
+            for handle, attribute in attributes.items():
+                if attribute['uuid'].hex[-4:] == '0328':
+                    success, result = self._read_handle(conn, handle, timeout)
+                    if not success:
+                        return False, None
+
+                    value = result['data']
+                    char = parse_characteristic_declaration(value)
+                    service['characteristics'][char['uuid']] = char
+                    last_char = char
+                elif attribute['uuid'].hex[-4:] == '0229':
+                    if last_char is None:
+                        return False, None
+
+                    success, result = self._read_handle(conn, handle, timeout)
+                    if not success:
+                        return False, None
+
+                    value = result['data']
+                    assert len(value) == 2
+                    value, = unpack("<H", value)
+
+                    last_char['client_configuration'] = {'handle': handle, 'value': value}
+
+        return True, {'services': services}
+
+    def _enable_rpcs(self, conn, services, timeout=1.0):
+        """Prepare this device to receive RPCs
```

Comment:
style - single line comment should end quotes on same line

Reply:
Fixed.